src/car.rs

```diff
@@ -0,0 +1,55 @@
+use {
+    super::Result,
+    crate::RacingCarError,
+    rand::Rng,
+    std::fmt,
+    std::fmt::{Display, Formatter},
+};
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct Car {
+    pub name: String,
+    pub distance: i32,
+}
+
+impl Car {
+    pub fn new(name: &str) -> Result<Self> {
+        if name.len() > 5 {
+            return Err(RacingCarError::InvalidCarName);
+        }
+        Ok(Car {
+            name: name.to_owned(),
+            distance: 0,
+        })
+    }
+
+    pub fn move_forward(&mut self) {
+        let rand_num = rand::thread_rng().gen_range(0..=10);
```

Comment:
`rand_num`은 우리가 제어할 수 없는 변수이기 때문에, 테스트코드를 작성할 수 없으니
함수의 파라미터로 빼서 테스트하기 용이하게 만들고, random에 대한 책임은 `move_forward`를 호출하는 친구한테 넘긴다.

Reply:
[translate]

Since `rand_num` is a variable that we cannot control, we cannot write test code.
It makes it easier to test by subtracting it as a parameter of the function, and passes the responsibility for random to the friend who calls `move_forward`.